sdks/java/io/bigquery-io-perf-tests/src/test/java/org/apache/beam/sdk/bigqueryioperftests/BigQueryIOReadPerformanceIT.java

```diff
@@ -0,0 +1,205 @@
+    options = IOITHelper.readIOTestPipelineOptions(BigQueryPerfTestOptions.class);
+    tempRoot = options.getTempRoot();
+    sourceOptions =
+        SyntheticOptions.fromJsonString(options.getSourceOptions(), SyntheticSourceOptions.class);
+    bigQueryMetricsDataset = options.getBigQueryDataset();
+    bigQueryMetricsTable = options.getBigQueryTable();
+    bigQuerySourceDataset = options.getBigQuerySourceDataset();
+    bigQuerySourceTable = options.getBigQuerySourceTable();
+    BigQueryOptions bigQueryOptions = BigQueryOptions.newBuilder().build();
+    tableQualifier =
+        String.format(
+            "%s:%s.%s", bigQueryOptions.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
+    if (!checkForSourceTable()) {
+      createAndFillSourceTable();
+    }
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    BigQueryOptions options = BigQueryOptions.newBuilder().build();
+    BigQuery client = options.getService();
+    TableId tableId =
+        TableId.of(options.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
+    client.delete(tableId);
+  }
+
+  private static void createAndFillSourceTable() {
+
+    Pipeline pipeline = Pipeline.create();
+
+    pipeline
+        .apply("Read from source", Read.from(new SyntheticBoundedSource(sourceOptions)))
+        .apply("Map records", ParDo.of(new MapKVToV()))
+        .apply(
+            "Write to BQ",
+            BigQueryIO.<byte[]>write()
+                .to(tableQualifier)
+                .withFormatFunction(
+                    input -> {
+                      TableRow tableRow = new TableRow();
+                      tableRow.set("data", input);
+                      return tableRow;
+                    })
+                .withCustomGcsTempLocation(ValueProvider.StaticValueProvider.of(tempRoot))
+                .withSchema(
+                    new TableSchema()
+                        .setFields(
+                            Collections.singletonList(
+                                new TableFieldSchema().setName("data").setType("BYTES")))));
+    pipeline.run().waitUntilFinish();
+  }
+
+  private static boolean checkForSourceTable() {
+    BigQueryOptions options = BigQueryOptions.newBuilder().build();
+    BigQuery client = options.getService();
+    TableId tableId =
+        TableId.of(options.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
+    Table sourceTable = client.getTable(tableId, TableOption.fields(BigQuery.TableField.TYPE));
+    return sourceTable != null;
+  }
+
+  @Test
+  public void testRead() {
+    Pipeline pipeline = Pipeline.create(options);
+    pipeline
+        .apply("Read from BQ", BigQueryIO.readTableRows().from(tableQualifier))
+        .apply("Gather time", ParDo.of(new TimeMonitor<>(NAMESPACE, "read_time")));
+    PipelineResult result = pipeline.run();
+    result.waitUntilFinish();
+    Collection<NamedTestResult> testResults = extractReadMetrics(getMetricSupplier(), result);
```

Comment:
I think we can do

```
supplier.apply() 
```

on the Function itself without wrapping this in the list (which is always a singleton list now btw). This will make `extractReadMetrics` metod redundant

Reply:
Good point, I changed it. Is it what you had in mind?